task/SSUP-34-Add-Endgame-popup

casey-gregor opened the pull request.

casey-gregor committed:
Created EndGamePopup and prefabs.
Applied changes to GameScene.unity
Commented out loading and reloading od scene
Added TimeScaleResetter and added it to MenuScene and GameScene installers

casey-gregor committed:
removed unnecessary classes

PutienkoEN commented on Assets/Game/Modules/Level/Scripts/LevelEventManager.cs:
```diff
@@ -45,5 +46,10 @@ private List<IGameEventHandler> GetHandlers(List<ILevelEventData> gameLevelEvent
                 .ToList();
         }
+
+        public void OnGameFinish()
```
Not used. Let's remove

PutienkoEN commented on Assets/Game/Modules/GameSpeed/Scripts/TimeScaleResetter.cs:
```diff
@@ -0,0 +1,13 @@
+
+namespace Game.Modules.GameSpeed
+{
+    public class TimeScaleResetter
```
Not used. Let's remove.

PutienkoEN commented on Assets/Game/Modules/UImodule/Scripts/EndGamePopupController.cs:
```diff
@@ -0,0 +1,48 @@
+    public sealed class EndGamePopupController : IInitializable, IDisposable, IGameFinishListener
+    {
+        private readonly EndGamePopupView _endGamePopupView;
+        private const string MAIN_MENU_SCENE = "MenuScene"; //ToDo : Will be used when FinishGame is fixed;
```
Better use GameSceneManager, and hide all scenes there.

casey-gregor committed:
fixed pr comments

casey-gregor committed:
Merge remote-tracking branch 'origin/dev' into task/SSUP-34-Add-Endgame-popup

# Conflicts:
#	Assets/Game/LifeCycle/Prefabs/[GameSceneContext].prefab
#	Assets/Game/Modules/Level/Scripts/LevelEventManager.cs
#	Assets/Game/Modules/MainMenu/Game.Modules.MainMenu.asmdef
#	Assets/Game/Scenes/GameScene.unity

casey-gregor committed:
merge fixes

casey-gregor committed:
removed unnecessary using in GameSceneInstaller.cs and PauseGameFinishListener.cs
fixed DOTween error in GameSpeedManager.cs, when scene reloaded
Added TimeScaleResetter.cs to MainMenuInstaller to fix timescale when loading the scene after gameover.
Small edits in EndGamePopupController.cs
Edited Bullet prefab to not play the shooting sound on Awake
Added _fireDelayStart in WeaponComponent.cs, so that shooting
starts with a defined delay.
Renamed UI folder to UIElements
Changed DOTWeen settings to display Errors
